kubechain/api/v1alpha1/mcpserver_types.go

```diff
@@ -0,0 +1,128 @@
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// MCPServerSpec defines the desired state of MCPServer
+type MCPServerSpec struct {
+	// Type specifies the transport type for the MCP server
+	// +kubebuilder:validation:Enum=stdio;http
+	// +kubebuilder:validation:Required
+	Type string `json:"type"`
+
+	// Command is the command to run for stdio MCP servers
+	// +optional
+	Command string `json:"command,omitempty"`
+
+	// Args are the arguments to pass to the command for stdio MCP servers
+	// +optional
+	Args []string `json:"args,omitempty"`
+
+	// Env are environment variables to set for stdio MCP servers
+	// +optional
+	Env []EnvVar `json:"env,omitempty"`
+
+	// URL is the endpoint for HTTP MCP servers
+	// +optional
+	URL string `json:"url,omitempty"`
+
+	// ResourceRequirements defines CPU/Memory resources requests/limits
+	// +optional
+	Resources ResourceRequirements `json:"resources,omitempty"`
```

Comment:
this makes sense for now, I think someday we might  want to expose the full [PodTemplateSpec](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-template-v1/)

Reply:
although I guess there's an argument that "if you wanna customize the runtime, just put it in a container that YOU write and use SSE/https"